Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -432,6 +439,73 @@ void TextureMapperLayer::collectDamageSelf(TextureMapperPaintOptions& options, D
     m_damage = Damage();
 }
 
+void TextureMapperLayer::collectDamageSelfChildrenReplicaFilterAndMask(TextureMapperPaintOptions& options, Damage& damage)
+{
+    bool hasFilterOrMask = [&] {
+        if (hasFilters())
+            return true;
+        if (m_state.maskLayer)
+            return true;
+        if (m_state.replicaLayer && m_state.replicaLayer->m_state.maskLayer)
+            return true;
+        return false;
+    }();
```

Comment:
What's the value of using a lambda in this case? The following looks shorter and easier to read to me:


```suggestion
    const bool hasFilterOrMask = hasFilters() || m_state.maskLayer ||
        (m_state.replicaLayer && m_state.replicaLayer->m_state.maskLayer);
```